static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -94,46 +122,106 @@ const setProgressTracker = (atomId: string): IntervalID => {
     return players[atomId].progressTracker;
 };
 
-const onPlayerPlaying = (atomId: string): void => {
-    const player = players[atomId];
-    const currentVideo = player.player.getVideoUrl().split('?v=')[1];
-    const originalVideo = player.iframe.dataset.assetId;
-
-    if (!player) {
-        return;
-    }
+const handlePlay = (atomId: string, player: AtomPlayer): void => {
+    const { trackingId, iframe, overlay, endSlate } = player;
 
     killProgressTracker(atomId);
+    setProgressTracker(atomId);
 
-    // TODO: implement progress tracking for related videos
-    if (currentVideo === originalVideo) {
-        setProgressTracker(atomId);
-        trackYoutubeEvent('play', getTrackingId(atomId));
+    // don't track play if resumed from a paused state
+    if (player.paused) {
+        player.paused = false;
+    } else {
+        trackYoutubeEvent('play', trackingId);
     }
 
-    const mainMedia =
-        (player.iframe && player.iframe.closest('.immersive-main-media')) ||
-        null;
-    const parentNode = player.overlay && player.overlay.parentNode;
-    const endSlateContainer =
-        parentNode && parentNode.querySelector('.end-slate-container');
+    const mainMedia = iframe.closest('.immersive-main-media');
 
     if (mainMedia) {
         mainMedia.classList.add('atom-playing');
     }
 
-    if (player.endSlate && !endSlateContainer) {
-        player.endSlate.fetch(parentNode, 'html');
+    if (overlay && endSlate) {
+        const parentElem = overlay.parentElement;
+
+        if (parentElem) {
+            endSlate.fetch(parentElem, 'html');
+        }
+    }
+};
+
+const getYoutubeIdFromUrl = (url: string): string => {
+    const youtubeIdKey = 'v';
+    const splitUrl = url.split('?');
+
+    if (splitUrl.length === 1) {
+        return '';
+    }
+
+    const queryParams = getUrlVars(splitUrl[1]);
+
+    return queryParams[youtubeIdKey] || '';
+};
+
+const onPlayerPlaying = (atomId: string): void => {
+    const player = players[atomId];
+
+    if (!player) {
+        return;
+    }
+
+    const { youtubePlayer, youtubeId } = players[atomId];
+
+    /**
+     * Get the youtube video id from the video currently playing.
+     * We want to compare with the youtube ID in memory. If they differ
+     * a related video has begun playing, so we need to get the atom ID
+     * for tracking.
+     */
+    const latestYoutubeId = getYoutubeIdFromUrl(youtubePlayer.getVideoUrl());
```

Comment:
I think the atom id is [on the DOM](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/atoms/youtube.scala.html#L44) as a data attribute, could we use that to avoid the `fetch`?

Reply:
The atom ID of the original video is in the DOM, once a related video plays this atom ID no longer matches that of the active video which, we want to track. The only way we can get the correct atom ID is to make this fetch call passing the youtube ID of the active video.